fennel/client/client.py

```diff
@@ -759,6 +759,34 @@ def inspect(
 
         return self._parse_dataframe(result, output_dtypes)
 
+    # ----------------------- Secret API's -----------------------------------
+
+    def get_secret(self, secret_name: str) -> Optional[str]:
+        response = self._get("{}/secret/{}".format(V1_API, secret_name))
+        if response.status_code != requests.codes.OK:
+            raise Exception(response.json())
+        return response.json().get("value")
+
+    def add_secret(self, secret_name: str, secret_value: str):
+        data = {"value": secret_value}
+        response = self._post_json(
+            "{}/secret/{}".format(V1_API, secret_name),
+            data,
+        )
+        if response.status_code != requests.codes.OK:
+            raise Exception(response.json())
+        return response
+
+    def update_secret(self, secret_name: str, secret_value: str):
```

Comment:
lets add a delete secret too ?

Reply:
Abhay suggested to have secrets immutable, should we allow users to delete secrets?

Reply:
Added the capability, basically soft deleted the mapping to execute the same.